app/src/main/java/com/yml/womensafety/navigationdrawer/home/HomePageActivity.kt

```diff
@@ -52,5 +98,92 @@ class HomePageActivity : AppCompatActivity() {
                 }
             }
         }
+        btn_fab.setOnClickListener {
+            sendSms()
+        }
+    }
+
+    //This method will send the SMS along with location
+    private fun sendSms() {
+        contactsViewModel.getContactsList(object : ContactsResponse {
+            override fun onContactsReceiveSuccess(userContacts: List<String>) {
+                try {
+                    val gpsTracker = GPSTracker(applicationContext)
```

Comment:
Can we move the business logic in this method to view model or so? View should now have contain the business logic in MVVM

Reply:
Done

Reply:
Done